apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -275,48 +260,131 @@ class PageTemplateModal extends Component {
 	}
 
 	getTemplateGroups = () => {
-		return {
-			blankTemplate: filter( this.props.templates, { slug: 'blank' } ),
-			currentTemplate: filter( this.props.templates, { slug: 'current' } ),
-			aboutTemplates: filter( this.props.templates, { category: 'about' } ),
-			blogTemplates: filter( this.props.templates, { category: 'blog' } ),
-			contactTemplates: filter( this.props.templates, { category: 'contact' } ),
-			eventTemplates: filter( this.props.templates, { category: 'event' } ),
-			menuTemplates: filter( this.props.templates, { category: 'menu' } ),
-			portfolioTemplates: filter( this.props.templates, { category: 'portfolio' } ),
-			productTemplates: filter( this.props.templates, { category: 'product' } ),
-			servicesTemplates: filter( this.props.templates, { category: 'services' } ),
-			teamTemplates: filter( this.props.templates, { category: 'team' } ),
-			homepageTemplates: sortBy( filter( this.props.templates, { category: 'home' } ), 'title' ),
-			comingSoonTemplates: filter( this.props.templates, { category: 'coming-soon' } ),
-		};
+		if ( ! this.props.templates.length ) {
+			return null;
+		}
+
+		const templateGroups = {};
+		for ( const template of this.props.templates ) {
+			for ( const key in template.categories ) {
+				if ( ! ( key in templateGroups ) ) {
+					templateGroups[ key ] = template.categories[ key ];
+				}
+			}
+		}
+
+		return this.sortGroupsNames( templateGroups );
+	};
+
+	sortGroupsNames = ( groups ) => {
+		return Object.keys( groups )
+			.sort()
+			.reduce( ( result, key ) => {
+				result[ key ] = groups[ key ];
+				return result;
+			}, {} );
+	};
+
+	getTemplatesForGroup = ( groupName ) => {
+		if ( ! this.props.templates.length ) {
+			return null;
+		}
+
+		if ( 'blank' === groupName ) {
+			return [ { name: 'blank', title: 'Blank', html: '' } ];
+		}
+
+		if ( 'current' === groupName && '' !== this.props._starter_page_template ) {
+			for ( const template of this.props.templates ) {
+				if ( this.props._starter_page_template === template.name ) {
+					return [ template ];
+				}
+			}
+		}
+
+		const templates = [];
+		for ( const template of this.props.templates ) {
+			for ( const key in template.categories ) {
+				if ( key === groupName ) {
+					templates.push( template );
+				}
+			}
+		}
+
+		return templates;
+	};
+
+	renderTemplateGroups = () => {
+		const unfilteredGroups = this.getTemplateGroups();
+		const groups = ! this.props.isFrontPage
+			? unfilteredGroups
+			: omit( unfilteredGroups, 'home-page' );
+
+		if ( ! groups ) {
+			return null;
+		}
+
+		const currentGroup =
+			'blank' !== this.props._starter_page_template
+				? this.renderTemplateGroup( 'current', __( 'Current', 'full-site-editing' ) )
+				: null;
+
+		const blankGroup = this.renderTemplateGroup( 'blank', __( 'Blank', 'full-site-editing' ) );
+
+		const homePageGroup = this.props.isFrontPage
+			? this.renderTemplateGroup( 'home-page', __( 'Home Page', 'full-site-editing' ) )
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **66** times:
`translate( 'Home page' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/patterns-api-to-template-selector)